src/self-managed-debug/src/system_catalog_dumper.rs

```diff
@@ -206,3 +217,577 @@ pub fn spawn_sql_port_forwarding_process(
         }
     })
 }
+
+#[derive(Debug, Clone)]
+pub enum RelationCategory {
+    /// For relations that belong in the `mz_introspection` schema.
+    /// These relations require a replica name to be specified.
+    Introspection,
+    /// For relations that are retained metric objects that we'd also like to get the SUBSCRIBE output for.
+    Retained,
+    /// Other relations that we want to do a SELECT * FROM on.
+    Basic,
+}
+
+#[derive(Debug, Clone)]
+pub struct Relation {
+    pub name: &'static str,
+    pub category: RelationCategory,
+}
+
+const RELATIONS: &[Relation] = &[
+    // Basic object information
+    Relation {
+        name: "mz_audit_events",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_databases",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_schemas",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_tables",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_sources",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_sinks",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_views",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_materialized_views",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_secrets",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_connections",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_roles",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_subscriptions",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_object_fully_qualified_names",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_sessions",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_object_history",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_object_lifetimes",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_object_dependencies",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_object_transitive_dependencies",
+        category: RelationCategory::Basic,
+    },
+    // Compute
```

Comment:
ohhhh I see now, these comments categorize the types of relations we're querying. Mind calling _that_ out in a doc comment so when folks add new relations they know they're sorted?